executor/linux/secret.go

```diff
@@ -65,37 +63,11 @@ func (s *secretSvc) create(ctx context.Context, ctn *pipeline.Container) error {
 		return err
 	}
 
-	logger.Debug("marshaling configuration")
-	// marshal container configuration
-	body, err := json.Marshal(ctn)
+	logger.Debug("substituting container configuration")
```

Comment:
I'm not really sure what's going on with the changes to this file. 
Unrelated - also looks like on line 48 there is a ref to a hard coded vela version.

Reply:
> I'm not really sure what's going on with the changes from json.Marshal(ctn) -> ctn.substitute()

Yeah that's my fault 😞 

That is a continuation of https://github.com/go-vela/pkg-executor/pull/86 which I missed in the original PR.

More information can be found here:

https://github.com/go-vela/types/pull/129

But tl;dr is that we moved this chunk of logic into [go-vela/types](https://github.com/go-vela/types)

> Unrelated to that - also looks like on line 48 there is a ref to a hard coded vela version. Should that be changed too?

Yeah at some point that has to change but I haven't come up with a way to refactor that yet.

Hopefully in time, I'll have an idea on how we can fix that 😅

Reply:
thanks for the info!